src/franka_plan_runner.cc

```diff
@@ -591,8 +592,40 @@ void FrankaPlanRunner::IncreaseFrankaTimeBasedOnStatus(
     // } else if (status_ == RobotStatus::Reversing) {
     //   // walk back in time
     //   franka_time_ -= period_in_seconds;
-  } else if (status_ == RobotStatus::Paused) {
+  }
+
+  if (status_ == RobotStatus::Paused) {
     // do nothing
+    if (cancel_plan_requested) {
+      if (plan_) {
```

Comment:
a cancellation request can happen when robot isn't in paused status, right?

Reply:
Yes but it will initiate the process of pausing, resulting in this code path being triggered. All cancelation requests will trigger a pause

Reply:
A cancellation is effectively a one-way pause (the original plan is discarded and cannot be resumed)